### PR TITLE
Feature/favorites cleanup

### PR DESCRIPTION
**Pull request by FodderMK**
This pull request more-or-less completes functionality for Favorites. I also added a common call to clear the console.

### CODE REVIEW

File: Packages/ca.tekly.favorites/Editor/Core/Data/Actions/FavoriteActionScriptAsset.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
+using Tekly.Common.Utils;
+using UnityEditor;
+using UnityEngine;
+
+namespace Tekly.Favorites
+{
+	[CreateAssetMenu(menuName = "Favorites/Scripted Action")]
+	public class FavoriteActionScriptAsset : FavoriteActionAsset
+	{
+		[SerializeReference] [Polymorphic] private IFavoriteActionScript m_referencedType;
```
Comment (matt-tekly) [inline Packages/ca.tekly.favorites/Editor/Core/Data/Actions/FavoriteActionScriptAsset.cs:10]: I didn't expect an Interface to work with this. I'm surprised it does.

File: Packages/ca.tekly.favorites/Editor/Core/Data/Actions/FavoriteActionMenu.cs [hunk 2]
```diff
@@ -1,15 +1,17 @@
...
-        public string Menu;
-        public override void Activate()
-        {
-            EditorApplication.ExecuteMenuItem(Menu);
-        }
-    }
+	[CreateAssetMenu(menuName = "Favorites/Menu Action")]
+	public class FavoriteActionMenu : FavoriteActionAsset
+	{
+		[SerializeField, FormerlySerializedAs("Menu")] private string m_Menu;
```
Comment (matt-tekly) [inline Packages/ca.tekly.favorites/Editor/Core/Data/Actions/FavoriteActionMenu.cs]: Should be named `m_menu`

File: Packages/ca.tekly.favorites/Editor/Core/Data/Actions/FavoriteActionScriptInspector.cs [hunk 3]
```diff
@@ -0,0 +1,24 @@
+using UnityEditor;
+using UnityEngine;
+
+namespace Tekly.Favorites
+{
+	[CustomEditor(typeof(FavoriteActionScriptAsset))]
+	public class FavoriteActionScriptEditor : UnityEditor.Editor
+	{
+		private int m_selectedIndex;
```
Comment (matt-tekly) [inline Packages/ca.tekly.favorites/Editor/Core/Data/Actions/FavoriteActionScriptInspector.cs]: Seems like these fields aren't used any longer

**Review by matt-tekly — CHANGES_REQUESTED** [review]

File: Packages/ca.tekly.favorites/Editor/Core/Data/Actions/FavoriteActionScriptInspector.cs (hunk 3 above)
Comment (FodderMK) [reply Packages/ca.tekly.favorites/Editor/Core/Data/Actions/FavoriteActionScriptInspector.cs]: Fixed

File: Packages/ca.tekly.favorites/Editor/Core/Data/Actions/FavoriteActionMenu.cs (hunk 2 above)
Comment (FodderMK) [reply Packages/ca.tekly.favorites/Editor/Core/Data/Actions/FavoriteActionMenu.cs]: Fixed

**Review by matt-tekly — APPROVED** [review]

**Pull request merged**